timemachine/fe/rbfe.py

```diff
@@ -393,7 +393,11 @@ def estimate_free_energy_given_initial_states(initial_states, protocol, temperat
     ukln_by_component_by_lambda = []
 
     for lamb_idx, initial_state in enumerate(initial_states):
-
+        # Clear any old references to avoid holding on to objects in memory we don't need.
+        cur_frames = None
+        cur_boxes = None
+        bound_impls = None
+        cur_batch_U_fns = None
```

Comment:
nit: alternative (maybe slightly clearer in intent)

```python
        del cur_frames, cur_boxes, bound_impls, cur_batch_U_fns
```

at the end of loop body